src/Git/CreateTagViaConsole.php

```diff
@@ -6,25 +6,39 @@
 
 use Laminas\AutomaticReleases\Git\Value\BranchName;
 use Laminas\AutomaticReleases\Gpg\SecretKeyId;
-use Psl\Env;
-use Psl\File;
-use Psl\Filesystem;
-use Psl\Shell;
+
+use function Psl\Env\temp_dir;
+use function Psl\File\write;
+use function Psl\Filesystem\create_temporary_file;
+use function Psl\Shell\execute;
 
 final class CreateTagViaConsole implements CreateTag
 {
+    public function __construct(private HasTag $hasTag)
+    {
+    }
+
     public function __invoke(
         string $repositoryDirectory,
         BranchName $sourceBranch,
         string $tagName,
         string $changelog,
         SecretKeyId $keyId,
     ): void {
-        $tagFileName = Filesystem\create_temporary_file(Env\temp_dir(), 'created_tag');
+        if (($this->hasTag)($repositoryDirectory, $tagName) === true) {
+            return;
+        }
+
+        $tagFileName = create_temporary_file(temp_dir(), 'created_tag');
+
+        write($tagFileName, $changelog);
 
-        File\write($tagFileName, $changelog);
+        execute('git', ['checkout', $sourceBranch->name()], $repositoryDirectory);
 
-        Shell\execute('git', ['checkout', $sourceBranch->name()], $repositoryDirectory);
-        Shell\execute('git', ['tag', $tagName, '-F', $tagFileName, '--cleanup=whitespace', '--local-user=' . $keyId->id()], $repositoryDirectory);
+        execute(
+            'git',
+            ['tag', $tagName, '-F', $tagFileName, '--cleanup=whitespace', '--local-user=' . $keyId->id()],
+            $repositoryDirectory,
+        );
```

Comment:
I think all this stays mostly unchanged, if the import changes are reverted/squashed